internal/providers/aws/aws.go

```diff
@@ -40,6 +40,11 @@ var (
 		Host:   "169.254.169.254",
 		Path:   "2019-10-01/user-data",
 	}
+	userdataBaseURL = url.URL{
```

Comment:
The name describes the URL but doesn't explain what we want with it.  Maybe something like `metadataServiceProbeURL`?